app/static/js/default.js

```diff
@@ -146,3 +149,16 @@ function setupDefaultChemiscope(j2sPath) {
         reader.readAsArrayBuffer(upload.files[0]);
     }
 }
+
+function download(filename, text) {
+  var element = document.createElement('a');
+  element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
```

Comment:
Since this is using a `data:` URL, we should check for URL size limit. I don't know if large files will be downloadable this way.

Reply:
could download a ~10MB json here. this whole concept is far from ideal though, e.g. hardcoded filename is crap. I let you change this based on best practices and your own instinct.

Reply:
Another options seems to be URL.createObjectURL(), which points to a `Blob` object, which is as unlimited as JS code goes (https://stackoverflow.com/a/43816041/4692076)

Code would look like this:
```js
const a = document.createElement('a');
a.download = filename;
a.href = URL.createObjectURL(new Blob([content], {type: mime}));
a.style.display = 'none';

document.body.appendChild(a);
a.click();
setTimeout(() => {
    document.body.removeChild(a);
    URL.revokeObjectURL(a.href);
}, 2000);
```

Reply:
Really, do as you see fit: I have no strong feeling at all.